src/Akeneo/CouplingDetector/Event/Events.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Akeneo\CouplingDetector\Event;
+
+/**
+ * List of event dispatched by the coupling detector.
+ *
+ * @author  Julien Janvier <j.janvier@gmail.com>
+ * @license http://opensource.org/licenses/MIT MIT
+ */
+final class Events
+{
+    /**
+     * The PRE_NODES_PARSED event occurs before nodes have been parsed.
+     *
+     * The event listener method receives a Akeneo\CouplingDetector\Event\PreNodesParsedEvent instance.
+     *
+     * @Event
+     */
+    const PRE_NODES_PARSED = 'php_cd.pre_nodes_parsed';
+
+    /**
+     * The NODE_PARSED event occurs when a node has been parsed.
+     *
+     * The event listener method receives a Akeneo\CouplingDetector\Event\NodeParsedEvent instance.
+     *
+     * @Event
+     */
+    const NODE_PARSED = 'php_cd.node_parsed';
+
+    /**
+     * The POST_NODES_PARSED event occurs when all nodes have been parsed.
+     *
+     * The event listener method receives a Akeneo\CouplingDetector\Event\PostNodesParsedEvent instance.
+     *
+     * @Event
+     */
+    const POST_NODES_PARSED = 'php_cd.post_nodes_parsed';
+
+    /**
+     * The PRE_RULES_CHECKED event occurs before all have been checked.
+     *
+     * The event listener method receives a Akeneo\CouplingDetector\Event\PreRulesCheckedEvent instance.
+     *
+     * @Event
+     */
+    const PRE_RULES_CHECKED = 'php_cd.pre_rules_checked';
```

Comment:
for a rule i would maybe say "matched" and not "checked", it imho seems more meaningful